packages/dev/loaders/src/BVH/bvhFileLoader.ts

```diff
@@ -0,0 +1,138 @@
+import type { ISceneLoaderPluginAsync, ISceneLoaderPluginFactory, ISceneLoaderPlugin, ISceneLoaderAsyncResult, ISceneLoaderPluginExtensions } from "core/Loading/sceneLoader";
+import { RegisterSceneLoaderPlugin } from "core/Loading/sceneLoader";
+import { AssetContainer } from "core/assetContainer";
+import type { Scene } from "core/scene";
+import { BVHLoader } from "./bvhLoader";
+import type { BVHLoadingOptions } from "./bvhLoadingOptions";
+import { BVHFileLoaderMetadata } from "./bvhFileLoader.metadata";
+
+declare module "core/Loading/sceneLoader" {
+    // eslint-disable-next-line jsdoc/require-jsdoc
+    export interface SceneLoaderPluginOptions {
+        /**
+         * Defines options for the bvh loader.
+         */
+        [BVHFileLoaderMetadata.name]: Partial<BVHLoadingOptions>;
+    }
+}
+
+/**
+ * @experimental
+ * BVH file type loader.
+ * This is a babylon scene loader plugin.
+ */
+export class BVHFileLoader implements ISceneLoaderPluginAsync, ISceneLoaderPluginFactory {
+    /**
+     * Name of the loader ("bvh")
+     */
+    public readonly name = BVHFileLoaderMetadata.name;
+
+    /** @internal */
+    public readonly extensions = BVHFileLoaderMetadata.extensions;
+
+    private _loadingOptions: BVHLoadingOptions;
+
+    /**
+     * Creates loader for bvh motion files
+     * @param loadingOptions - Options for the bvh loader
+     */
+    constructor(loadingOptions?: Partial<Readonly<BVHLoadingOptions>>) {
+        this._loadingOptions = { ...BVHFileLoader._DefaultLoadingOptions, ...(loadingOptions ?? {}) };
+    }
+
+    private static get _DefaultLoadingOptions(): BVHLoadingOptions {
+        return {
+            animationName: "Animation",
+            loopBehavior: 1,
+            skeletonName: "skeleton",
+            skeletonId: "skeleton_id",
+        };
+    }
+
+    /**
+     * Instantiates a bvh file loader plugin.
+     * @returns the created plugin
+     */
+    createPlugin(): ISceneLoaderPluginAsync | ISceneLoaderPlugin {
```

Comment:
Pass your options through the `createPlugin` call and into your constructor (see other loaders for an example). This will allow options to passed down through the top level scene loader functions.